internal/status/gateway.go

```diff
@@ -48,8 +49,15 @@ func prepareGatewayStatus(gatewayStatus state.GatewayStatus, transitionTime meta
 		})
 	}
 
+	ipAddrType := v1beta1.IPAddressType
```

Comment:
I think it's gonna be better (long term) if we parse all user input in one place -- the entry point  (main.go) -- and then propagate the required configuration options down to the components, similarly to what we do with cli arguments.  This way we will manage them all in one place.

Reply:
In this case you're referring to the POD_IP env var being set, right?

Reply:
> In this case you're referring to the POD_IP env var being set, right? Just clarifying since your comment is linked to the IPAddressType

yep, POD_IP